internal/k8s/client.go

```diff
@@ -42,12 +44,24 @@ func (cID ContainerID) ShortStr() string {
 
 func (nID NodeID) String() string { return string(nID) }
 
+func (n Namespace) String() string {
+	if n == "" {
+		return string(Default)
+	}
+	return string(n)
+}
+
 type Client interface {
 	Apply(ctx context.Context, entities []K8sEntity) error
 	Delete(ctx context.Context, entities []K8sEntity) error
 
-	PodWithImage(ctx context.Context, image reference.NamedTagged) (*v1.Pod, error)
-	PollForPodWithImage(ctx context.Context, image reference.NamedTagged, timeout time.Duration) (*v1.Pod, error)
+	PodWithImage(ctx context.Context, image reference.NamedTagged, n Namespace) (*v1.Pod, error)
+	PollForPodWithImage(ctx context.Context, image reference.NamedTagged, n Namespace, timeout time.Duration) (*v1.Pod, error)
+	PodByID(ctx context.Context, podID PodID, n Namespace) (*v1.Pod, error)
+
+	// Creates a channel where all changes to the pod are brodcast.
+	// Takes a pod as input, to indicate the version of the pod where we start watching.
```

Comment:
HTTP notification APIs typically take a "start watching at point X" parameter.

otherwise, you will have race conditions, because you have no idea how long it will take for the HTTP request to get to the server.

the "point X" parameter says to the server, "go back through your history, and send me any changes that happened since point X"